processing/src/main/java/org/apache/druid/query/aggregation/first/StringFirstAggregatorFactory.java

```diff
@@ -154,6 +160,26 @@ public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
     }
   }
 
+  @Override
+  public VectorAggregator factorizeVector(VectorColumnSelectorFactory selectorFactory)
+  {
+    ColumnCapabilities capabilities = selectorFactory.getColumnCapabilities(fieldName);
+    VectorObjectSelector vSelector = selectorFactory.makeObjectSelector(fieldName);
+    BaseLongVectorValueSelector timeSelector = (BaseLongVectorValueSelector) selectorFactory.makeValueSelector(
+        timeColumn);
+    if (capabilities != null) {
+      return new StringFirstVectorAggregator(timeSelector, vSelector, maxStringBytes);
+    } else {
+      return new StringFirstVectorAggregator(null, vSelector, maxStringBytes);
+    }
```

Comment:
We can/should do this a bit more intelligently.  Specifically, there are 3 different types of vector selectors that could be needed here and you will need to check column capabilities ahead of time to tell the difference:

1. If it is a STRING and multi-valued, use the multivalue-dimension version
2. If it is a STRING and single-valued, use the single value dimension version
3. Otherwise use a VectorObjectSelector

Your implementation for (3) is in this PR already, for (1) and (2), you can read only the dictionary ids and just keep track of only the earliest dictionaryId (not the string, the dictionary id).  Then, when `get()` is called, convert the dictionary id into the String and truncate the size if necessary.